grpc/stream.go

```diff
@@ -323,10 +329,19 @@ func (s *stream) end() {
 }
 
 func (s *stream) closeWithError(err error) error {
+	s.close(err)
+
+	return s.callErrorListeners(err)
```

Comment:
This means that a user might get multiple errors to be handled. Not certain if that is good or bad ...but I guess we can try it.